packages/ember-data/lib/system/store.js

```diff
@@ -1522,15 +1522,64 @@ Store = Service.extend({
   /**
     Push some data for a given type into the store.
 
-    This method expects normalized data:
+    This method expects normalized [JSON API](http://jsonapi.org/) document. This means you have to follow [JSON API specification](http://jsonapi.org/format/) with few minor adjustments:
+    - record's `type` should always be in singular, dasherized form
+    - members (properties) should be camelCased
 
-    * The ID is a key named `id` (an ID is mandatory)
-    * The names of attributes are the ones you used in
-      your model's `DS.attr`s.
-    * Your relationships must be:
-      * represented as IDs or Arrays of IDs
-      * represented as model instances
-      * represented as URLs, under the `links` key
+    [Your primary data should be wrapped inside `data` property](http://jsonapi.org/format/#document-top-level):
+
+    ```
```

Comment:
Lets add `js` after the back ticks so the markdown hi-lighter knows to use the JavaScript formatter instead of the generic one.